Modules/mathmodule.c

```diff
@@ -2761,26 +2761,49 @@ math_dist_impl(PyObject *module, PyObject *p, PyObject *q)
     return NULL;
 }
 
-/* AC: cannot convert yet, waiting for *args support */
+
+/*[clinic input]
+math.hypot
+
+    *coordinates as args: object
+
+Multidimensional Euclidean distance from the origin to a point.
+
+Roughly equivalent to:
+    sqrt(sum(x**2 for x in coordinates))
+
+For a two dimensional point (x, y), gives the hypotenuse
+using the Pythagorean theorem:  sqrt(x*x + y*y).
+
+For example, the hypotenuse of a 3/4/5 right triangle is:
+
+    >>> hypot(3.0, 4.0)
+    5.0
+
+[clinic start generated code]*/
+
 static PyObject *
-math_hypot(PyObject *self, PyObject *const *args, Py_ssize_t nargs)
+math_hypot_impl(PyObject *module, PyObject *args)
+/*[clinic end generated code: output=ace3a23188b6798e input=ff33f88cd8b1f06d]*/
 {
-    Py_ssize_t i;
+    Py_ssize_t i, nargs;
     PyObject *item;
     double max = 0.0;
     double x, result;
     int found_nan = 0;
     double coord_on_stack[NUM_STACK_ELEMS];
     double *coordinates = coord_on_stack;
 
+    nargs = PyTuple_GET_SIZE(args);
+
     if (nargs > NUM_STACK_ELEMS) {
         coordinates = (double *) PyObject_Malloc(nargs * sizeof(double));
         if (coordinates == NULL) {
             return PyErr_NoMemory();
         }
     }
     for (i = 0; i < nargs; i++) {
-        item = args[i];
+        item = PyTuple_GetItem(args, i);
```

Comment:
This might affect performance. Can you please measure it?

Reply:
Indeed, there is a difference.
```
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 10))' 'h(*a)'
500000 loops, best of 5: 650 nsec per loop
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 100))' 'h(*a)'
50000 loops, best of 5: 4.54 usec per loop
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 1000))' 'h(*a)'
5000 loops, best of 5: 45.1 usec per loop
$ git checkout fix-101123 
Switched to branch 'fix-101123'
Your branch is up to date with 'github/fix-101123'.
$ make -s
Checked 111 modules (30 built-in, 80 shared, 1 n/a on linux-x86_64, 0 disabled, 0 missing, 0 failed on import)
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 10))' 'h(*a)'
500000 loops, best of 5: 950 nsec per loop
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 100))' 'h(*a)'
50000 loops, best of 5: 6.43 usec per loop
$ ./python -m timeit -s 'from math import hypot as h' -s 'a = list(range(1, 1000))' 'h(*a)'
5000 loops, best of 5: 65.6 usec per loop
```

So, probably we should use the patch from the issue.  On another hand, the comment ("AC: cannot convert yet, waiting for *args support") - seems to be misleading and must be removed.

Reply:
You can use string-based docs to populate `__text_signature__`. It will fix your problem with `inspect.signature` with 0 runtime cost. Example from `dict.get`:


```c
PyDoc_STRVAR(dict_get__doc__,
"get($self, key, default=None, /)\n"
"--\n"
"\n"
"Return the value for key if key is in the dictionary, else default.");
```

Reply:
> You can use string-based docs to populate __text_signature__. 

Yes, I understand.  That seems to be the solution, presented in the [issue description](https://github.com/python/cpython/issues/101123), isn't?

The question was: should we keep the mentioned comment for math_hypot?  It seems to be misleading for me (issue reference would be more helpful here, if it is an open problem), as there is no principal problems to convert this function to the Argument Clinic.

Reply:
Yes, I think it should be changed to: `Not converted to AC because it affects performance`

Reply:
Could @rhettinger bless this change?  The [comment](https://github.com/python/cpython/blob/b84be8d9c0e6eca37be14c38250580251a3ef908/Modules/mathmodule.c#L2764) is coming from #8474 and I'm not sure he meant performance issues with AC (rather something like #64490).
PS: AC stuff reverted.